core/Db/BatchInsert.php

```diff
@@ -56,12 +56,26 @@ public static function tableInsertBatchIterate($tableName, $fields, $values, $ig
      */
     public static function tableInsertBatch($tableName, $fields, $values, $throwException = false)
     {
-        $filePath = StaticContainer::get('path.tmp') . '/assets/' . $tableName . '-' . Common::generateUniqId() . '.csv';
-
         $loadDataInfileEnabled = Config::getInstance()->General['enable_load_data_infile'];
 
         if ($loadDataInfileEnabled
             && Db::get()->hasBulkLoader()) {
+
+            try {
+                $path = Db::fetchOne('SELECT @@secure_file_priv'); // was introduced in 5.0.38
+            } catch (Exception $e) {
+                // we do not rethrow exception if $throwException = true as an error might be expected if < 5.0.38
+                // in this case tableInsertBatch might still work
+            }
+
+            if (empty($path) || !is_dir($path) || !is_writable($path)) {
```

Comment:
maybe you should also check `is_readable` as a folder can be writable yet not readable?